crosslink/internal/config.go

```diff
@@ -0,0 +1,57 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crosslink
+
+import (
+	"log"
+
+	"go.uber.org/zap"
+)
+
+type moduleInfo struct {
+	moduleFilePath            string
+	moduleContents            []byte
+	requiredReplaceStatements map[string]struct{}
+}
```

Comment:
The [`modfile.File`](https://pkg.go.dev/golang.org/x/mod/modfile#File) type can be used to contain all this information and has a full package devoted to the parsing and handling of that data. We recreate much of that functionality. Why not just use that type throughout this code instead?

Reply:
Sorry about taking so long to get back to these, I think I will have some time to refocus on Crosslink this week. You bring up a good point here about using `modfile.File`. I believe I overlooked or misunderstood what the first parameter in `modfile.Parse` was for. I now understand that it stores the file path for the applicable go.mod file. 

Knowing this I believe I could refactor out the `moduleFilePath` and `moduleContents` and instead store a single `modfile.File` object. I do not know off the top of my head if it makes sense to remove the `requiredReplaceStatements` because that is built up so that the file is only touched at modified at specific locations. I will look into that though and see what options exist.